src/framework/mpas_log.F

```diff
@@ -823,13 +839,15 @@ subroutine log_abort()
 
       ! Close the err log to be clean
       close(mpas_log_info % errorLog % unitNum)
+      call mpas_release_unit(mpas_log_info % errorLog % unitNum)
 
       deallocate(mpas_log_info % errorLog)
       deallocate(mpas_log_info % outputLog)
       deallocate(mpas_log_info)
 
 #ifdef _MPI
-      call MPI_Abort(MPI_COMM_WORLD, mpi_errcode, mpi_ierr)
+      if ( mpas_log_info % domainID == 0 ) &
+         call MPI_Abort(MPI_COMM_WORLD, mpi_errcode, mpi_ierr)
```

Comment:
Does this have any implications for situations in which domainID 0 does not encounter any fatal runtime issues, while some other domainID does?

Reply:
As this PR is currently written, domainID 0 is only used when MPI is initialized internally within MPAS, which does not allow more than one domain to be instantiated.  Thus domainID != 0 only occurs when mpi_comm is initialized externally, in which case we assume the external program will handle shutdown of MPI.  If we are to add the use-case of allowing multiple instances of domain to use an internally initialized MPI communicator, then this would need to change.  Maybe I could add a logical to domain and mpas_log_info such as "internal_mpi" that knows how MPI was initialized.  What do you think?